cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesStore.java

```diff
@@ -153,4 +168,15 @@ private String getMultipartKey(String... parts) {
     }
     return Bytes.toString(result);
   }
+
+  private void validateProfileExistsAndStatus(
```

Comment:
A better name would be `validateProperties` for better abstraction. Currently we only check for profile exists and status, but in future we could be checking for addition things.